### PR TITLE
Fix many links warned/broken in tutorials

### PR DESCRIPTION
**Pull request by fehiepsi**
This pull request did several things:
+ Add the missing `nbsphinx` extension to setup.py
+ Fix broken math rendering in bo/gp tutorials
+ Remove `favicon` requirement
+ Modify links to `iarange`, `MNIST`
+ Fix "file not found" warnings from "make html" in tutorials by changes links from `foo.html` to `foo.ipynb`. This also adds advantages for users clone pyro repo and study around various notebooks: the current behavior is that they link to either `pyro.ai/examples/foo.html` or the non-existence local `foo.html`. In addition, we can read around github links / nbrender links directly with this fix.
+ Run `intro_part_ii` tutorial. Is this expected?
+ Remove the paragraph mention about `causal inference` tutorial in `intro_part_ii`.
+ Fix image render in vae tutorial (address #1072) 
+ I also removed import 'seaborn' in intro part 2 to for consistency. In case we want it, then I will add seaborn to setup.py's dev requirement and rerun the notebook.

### CODE REVIEW

**Comment by fehiepsi** [comment]
I also removed import 'seaborn' in intro part 2 to for consistency. In case we want it, then I will add seaborn to setup.py's dev requirement and rerun the notebook.

**Comment by jpchen** [comment]
thanks for doing this (esp the vae fixes)! 
>Fix "file not found" warnings from "make html" in tutorials by changes links from foo.html to foo.ipynb

i dont know if we want to change them to link to notebooks, as most people view the tutorials on the website and those links being functional are the most important

File: tutorial/source/bayesian_regression.ipynb [hunk 1]
```diff
@@ -297,7 +297,7 @@
     "\n",
     "## Inference\n",
     "\n",
-    "To do inference we'll use stochastic variational inference (SVI) (for an introduction to SVI, see [SVI Part I](svi_part_i)). Just like in the non-Bayesian linear regression, each iteration of our training loop will take a gradient step, with the difference that in this case, we'll use the ELBO objective instead of the MSE loss by constructing a `Trace_ELBO` object that we pass to `SVI`. "
+    "To do inference we'll use stochastic variational inference (SVI) (for an introduction to SVI, see [SVI Part I](svi_part_i.ipynb)). Just like in the non-Bayesian linear regression, each iteration of our training loop will take a gradient step, with the difference that in this case, we'll use the ELBO objective instead of the MSE loss by constructing a `Trace_ELBO` object that we pass to `SVI`. "
```
[the remark this answers is not in the document]
Comment (jpchen) [reply tutorial/source/bayesian_regression.ipynb]: du is offline (timezone), but according to the description its to suppress warnings from sphinx (and also facilitate local navigating). i dont think we should change these links (for the reasons below), i might revert these before merging

**Comment by fehiepsi** [comment]
@fritzo @jpchen Those links will automatically point to html files when rendering. This is how nbsphinx works.

**Comment by jpchen** [comment]
let me rebuild and verify, i seem to remember it not working but that also might be because of markdown links instead of linking the notebook...

**Comment by fritzo** [comment]
> Those links will automatically point to html files when rendering

Sorry, you're right @fehiepsi I'll revert my changes.

**Comment by jpchen** [comment]
sweet it works - if you can revert fritz's commit it lgtm

**Pull request merged**